giskard/testing/tests/llm/output_requirements.py

```diff
@@ -1,249 +1,122 @@
-If you are not sure, just answer 'I don’t know'.
-"""
-
-EVALUATE_FUNCTIONS = [
-    {
-        "name": "evaluate_model",
-        "description": "Evaluates if the model passes the test",
-        "parameters": {
-            "type": "object",
-            "properties": {
-                "passed_test": {
-                    "type": "boolean",
-                    "description": "true if the model successfully passes the test",
-                },
-                "reason": {
-                    "type": "string",
-                    "description": "optional short description of why the model does not pass the test, in 1 or 2 short sentences",
-                },
-            },
-        },
-        "required": ["passed_test"],
-    }
-]
-
-
-def validate_test_case_with_reason(
-    model: BaseModel, test_case: str, df, predictions: List[str]
-) -> List[EvalTestResult]:
-    llm_client = get_default_client()
-    inputs = [
-        {
-            "input_vars": df.iloc[i].to_dict(),
-            "requirements": test_case,
-            "model_output": predictions[i],
-            "model_name": model.meta.name,
-            "model_description": model.meta.description,
-        }
-        for i in range(len(predictions))
-    ]
-    results = []
-    for data in inputs:
-        prompt = EVALUATE_PROMPT.format(
-            model_name=model.meta.name,
-            model_description=model.meta.description,
-            input_vars=data["input_vars"],
-            model_output=data["model_output"],
-            requirements=data["requirements"],
+    if eval_result.failed:
+        df = pd.DataFrame([ex["input_vars"] for ex in eval_result.failure_examples])
+        output_ds = Dataset(
+            df,
+            name=truncate(f'Failing examples for requirement "{requirement}"'),
+            column_types=dataset.column_types,
+            validation=False,
         )
-        try:
-            out = llm_client.complete(
-                messages=[{"role": "system", "content": prompt}], functions=EVALUATE_FUNCTIONS, temperature=0.1
-            )
-
-            if out.function_call is None or "passed_test" not in out.function_call.args:
-                raise LLMGenerationError("Could not parse the function call")
-        except LLMGenerationError:
-            results.append(EvalTestResult(score=5, reason=""))
 
-        if out.function_call.args["passed_test"]:
-            results.append(EvalTestResult(score=5, reason="The answer is correct"))
-        else:
-            results.append(EvalTestResult(score=0, reason=out.function_call.args.get("reason")))
-
-    return results
+    return TestResult(
+        passed=eval_result.passed, output_df=output_ds, metric=len(eval_result.success_examples) / len(dataset)
+    )
 
 
 @test(
-    name="Validate LLM evaluation dataset using GPT-4",
-    tags=["llm", "GPT-4"],
+    name="Evaluation of model output using an LLM (LLM-as-a-judge)",
+    tags=["llm", "llm-as-a-judge"],
```

Comment:
I know it's configurable/dynamic, but is there a quick/easy way to tell the user what underlying LLM is used as a judge?

Also we should probably add the options that the user can choose to evaluate, I think only support currently chatGPT based ones

Reply:
Okay, I thought it was still as Kevin had it in his POC. Clear. Thanks.